tests/test_core.py

```diff
@@ -199,26 +196,110 @@ class test_TypeDict(unittest.TestCase):
 
     def test_init(self):
         """Test construction with different list types."""
-        raise NotImplementedError()
+        types = ('A','B')
+        default = {'A':1.0, 'B':1.0}
+
+        #test construction with tuple input
+        d = relentless.core.TypeDict(types=('A','B'))
+        self.assertEqual(d.types, types)
+
+        #test construction with list input
+        d = relentless.core.TypeDict(types=['A','B'])
+        self.assertEqual(d.types, types)
+
+        #test construction with defined default input
+        d = relentless.core.TypeDict(types=('A','B'), default=1.0)
+        self.assertEqual(d.types, types)
+        self.assertEqual(d._data, default)
+
+        types = ('A',)
```

Comment:
This variable looks out of place here, can you place it with the test below your comment?